checkstyle/checkstyle.xml

```diff
@@ -34,7 +34,14 @@
   <module name="TreeWalker">
 
     <!-- code cleanup -->
-    <module name="UnusedImports"/>
+    <module name="UnusedImports">
+      <property name="id" value="allowJavadocUnusedImports"/>
+      <property name="processJavadoc" value="true"/>
```

Comment:
That's nice that this granularity is available. What do you think about just enabling this flag for all modules and deleting the "preventJavadocUnusedImports" rule/supression?
From my understanding, if you have rules for processJavadoc as true and false, then the stricter one just always applies anyways, so this new "allowJavadocUnusedImports" doesn't actually do anything right now. The change that is allowing the build to pass is that you did the suppression.

Reply:
For normal classes it's an AND of strict and relaxed, so it's strict. For samza-api, I disabled strict specifically, so it's relaxed. I wanted to make a minimal change from existing behavior, but I'm in favor of relaxing this check everywhere too.

@nickpan47 WDYT? Are you OK with relaxing checkstyle unused imports check in Samza such that it adding imports for reference in Javadocs?

Reply:
Ah right, I had a feeling that I was missing something. If you didn't add the "allowJavadocUnusedImports", then samza-api would have no unused imports checks at all, but we still would want a check for fully unused imports.